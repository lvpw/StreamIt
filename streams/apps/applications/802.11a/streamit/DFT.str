/* extremely simple N^2 algorithm for calculating N point DFT */

complex->complex splitjoin DFT(int N) {

  split duplicate;

  for(int i=0; i<N; i++)
    add calc_kth_coeff(N,i); 

  join roundrobin(1);
}


complex->complex pipeline IDFT(int N) {

  add complex_conjugate();
  add DFT(N);
  add complex_conjugate();
  add divide_by_N(N);

}


complex->complex pipeline calc_kth_coeff(int N, int k) {

  add splitjoin {
    split roundrobin(N,0);
    add Identity<complex>;
    add create_unity_train(N,k);
    join roundrobin(1,1);
  }

  add mult_and_sum(N);

} 



complex->complex filter mult_and_sum(int N) {

  work push 1 pop 2*N {

    complex total = 0.0;

    for(int i=0; i<N; i++)
      total += pop()*pop();

    push(total);

  }
}


void->complex filter create_unity_train(int N, int k) {

  complex temp;

  init {
    complex num = 0.0 - 1.0i;
    temp = 2*pi*num*k/N;
  }
  
  work push N pop 0 {

    for(int i=0; i<N; i++) 
      push(exp(temp*i));

  }
}


complex->complex filter divide_by_N(int len) {

  work push len pop len {

    for(int i=0; i<len; i++)
      push(pop()/len);
  }
}



complex->complex filter complex_conjugate() {

  work push 1 pop 1 {

    complex c = pop();
    c.imag = -c.imag;
    push(c);
  }
}
