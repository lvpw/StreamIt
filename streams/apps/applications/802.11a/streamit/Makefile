# import the FFT without the void->void section
all: ../../../benchmarks/fft/streamit/FFT6.str
	./importFFT.pl > FFT6.str
