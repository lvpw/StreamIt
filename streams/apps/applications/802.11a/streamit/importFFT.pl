#!/usr/local/bin/perl

# Writes out all of FFT6 except for the void->void stream.  We need to
# do this since strc expects only one void->void stream in the input
# files.

open(MYFILE, "../../../benchmarks/fft/streamit/FFT6.str") or die("Cannot find FFT6.str.\n");
{
    local $/;
    $text = <MYFILE>;
}

$text =~ s!void->void.*complex->complex filter CombineDFT!complex->complex filter CombineDFT!is;

print $text;
close(MYFILE);
