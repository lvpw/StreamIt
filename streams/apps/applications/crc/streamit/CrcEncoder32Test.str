/**
 *   CRCEncoder32:
 *   CRC Encoder using a 32 bit
 *   Generator Polynomial
 *   NOTE:  Until finite input lengths are accepted by StreamIt,
 *          this will not completely work as expected.
 *   - J. Wong
 *
 *  (Bill translated to new syntax, converted state to peeking)
 */

int->int filter AdditionFilter {
    work push 2 pop 2 {
	int mInitFeedbackData = pop();
	int mShiftRegisterInput = pop();
        int mAdditionOutput;
	
	if(((mInitFeedbackData == 1) && (mShiftRegisterInput == 1)) || ((mInitFeedbackData == 0) && (mShiftRegisterInput == 0)))
	{
	    mAdditionOutput = 0;
	}
	else
	{
	    mAdditionOutput = 1;    
	} 
	push (mInitFeedbackData);
	push (mAdditionOutput);
    }
}

int->int filter InitialAdditionFilter {
    work push 2 pop 2 {
	int mFileInput = pop();
	int mInitFeedbackData = pop();
        int mAdditionOutput;
	//System.err.println("File Input is " + mFileInput + " Feedback data is " + mInitFeedbackData);
	if(((mInitFeedbackData == 1) && (mFileInput == 1)) || ((mInitFeedbackData == 0) && (mFileInput == 0))) {
	    mAdditionOutput = 0;
	}
	else {
	    mAdditionOutput = 1;    
	} 
	push(mAdditionOutput);
	push(mAdditionOutput);
    }

}

/*
int->int filter ShiftRegisterFilter() {
    int mRegisterContents;
    
    work push 2 pop 2 {
	int mInitFeedbackData = pop();
	int mDataStream = pop();
	int mRegisterOutput = mRegisterContents;   //move datastream value into Shift Register
// 	mRegisterContents = mDataStream;       //move original contents to output
// 	mTerminationCounter++;
// 	if(mTerminationCounter < EncoderInput.mInputLength + 1)
// 	{
// 	    System.err.println("Shift Register Contents of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterContents);
// 	    System.err.println("Shift Register Output of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterOutput);
// 	}
	push(mInitFeedbackData);
	push(mRegisterOutput);
    }
}
*/

// state eliminated in previous filter in favor of peeking
int->int filter ShiftRegisterFilter() {
    prework push 2 pop 1 {
	push(pop());
	push(0);
    }
    
    work push 2 pop 2 {
	int mRegisterContents = pop();
	int mInitFeedbackData = pop();
	push(mInitFeedbackData);
	push(mRegisterContents);
    }
}

void->int filter CrcInputFilter {
    // int[] mdata;
    boolean counter;
    // EncoderInput mencodefile;

    work push 1 {
	if(!counter)
	{
	    counter = true;
	    push (0);
	}
	else
	{
	    push (1);
	}     
    }
}

int->int filter FeedbackEndFilter {
    work pop 2 push 1 {
	int mInitFeedbackData = pop();
	int mRegisterOutput = pop();
	//System.err.println("Shift Register 3 is " + mRegisterOutput);
	push(mRegisterOutput);
    }
}

int->void filter IntPrinter {
    work pop 1 {
	println(pop());
    }
}

int->int feedbackloop CrcFeedbackLoop {
    enqueue(0);
    join roundrobin();
    body FeedbackBodyStream();
    split duplicate;
    loop Identity<int>();
}

int->int pipeline FeedbackBodyStream {
    add InitialAdditionFilter();
    add ShiftRegisterFilter();
    add AdditionFilter();
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^2
    add ShiftRegisterFilter();
    //x^3
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^4
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^5
    add ShiftRegisterFilter();
    //x^6
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^7
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^8
    add ShiftRegisterFilter();
    //x^9
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^10
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^11
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^12
    add ShiftRegisterFilter();
    //x^13
    add ShiftRegisterFilter();
    //x^14
    add ShiftRegisterFilter();
    //x^15
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^16
    add ShiftRegisterFilter();
    //x^17
    add ShiftRegisterFilter();
    //x^18
    add ShiftRegisterFilter();
    //x^19
    add ShiftRegisterFilter();
    //x^20
    add ShiftRegisterFilter();
    //x^21
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^22
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^23
    add ShiftRegisterFilter();
    //x^24
    add ShiftRegisterFilter();
    //x^25
    add ShiftRegisterFilter();
    add AdditionFilter(); //x^26
    add ShiftRegisterFilter();
    //x^27
    add ShiftRegisterFilter();
    //x^28
    add ShiftRegisterFilter();
    //x^29
    add ShiftRegisterFilter();
    //x^30
    add ShiftRegisterFilter();
    //x^31
    add Identity<int>();

    //add ShiftRegisterFilter(3);
    add FeedbackEndFilter();
}

void->void pipeline CrcEncoder32Test {
    add CrcInputFilter();
    add CrcFeedbackLoop();
    add IntPrinter();
}
