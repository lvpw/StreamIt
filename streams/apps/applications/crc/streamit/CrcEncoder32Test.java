/**
 *   CRCEncoder32.java:
 *   CRC Encoder using a 32 bit
 *   Generator Polynomial
 *   NOTE:  Until finite input lengths are accepted by StreamIt,
 *          this will not completely work as expected.
 *   - J. Wong
 */

import streamit.library.*;
import java.lang.*;
import streamit.library.io.*;
import java.io.*;

/*
class EncoderInput
{

    //member variables!
    public static int mInputLength = 5;


    public int[] readFile() 
  {
    int[] input = new int[mInputLength];
    try
    {
	File f1 = new File("CrcEncoderInput1");
	java.io.FileReader fr = new java.io.FileReader(f1);
	BufferedReader br = new BufferedReader(fr);
	//DataInputStream data = new DataInputStream(new FileInputStream(f1));
	//read the sucker!
	//int[] input = new int[4];
	for(int i = 0; i < input.length; i++)
	    {
		String j = br.readLine();
		
		if (j.equals("1"))
		    {
			input[i] = 1;
		    }
		else
		    { 
			input[i] = 0;
		    }		
		
	    }
	br.close();
    }
    catch (IOException e)
    {
	System.out.println("Blah! IO error!");
    }

    return input;
  }//readFile

}//class EncoderInput
*/

class AdditionFilter extends Filter
{
    //inputs:
    int mInitFeedbackData;
    int mShiftRegisterInput;
    //outputs
    //output 1: mInitFeedbackData
    int mAdditionOutput;


    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
    }

    public void work()
    {
	mInitFeedbackData = input.popInt();
	mShiftRegisterInput = input.popInt();
	
	if(((mInitFeedbackData == 1) && (mShiftRegisterInput == 1)) || ((mInitFeedbackData == 0) && (mShiftRegisterInput == 0)))
	{
	    mAdditionOutput = 0;
	}
	else
	{
	    mAdditionOutput = 1;    
	} 
	output.pushInt(mInitFeedbackData);
	output.pushInt(mAdditionOutput);
    }

}//AdditionFilter

class InitialAdditionFilter extends Filter

{
    //inputs:
    int mInitFeedbackData;
    int mFileInput;
    //outputs
    //output 1: mInitFeedbackData
    int mAdditionOutput;


    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
    }

    public void work()
    {
	mFileInput = input.popInt();
	mInitFeedbackData = input.popInt();
	//System.err.println("File Input is " + mFileInput + " Feedback data is " + mInitFeedbackData);
	if(((mInitFeedbackData == 1) && (mFileInput == 1)) || ((mInitFeedbackData == 0) && (mFileInput == 0)))
	{
	    mAdditionOutput = 0;
	}
	else
	{
	    mAdditionOutput = 1;    
	} 
	output.pushInt(mAdditionOutput);
	output.pushInt(mAdditionOutput);
    }

}//InitialAdditionFilter

class ZeroAdditionFilter extends Filter
{
    //inputs:
    int mInitFeedbackData;
    int mShiftRegisterInput;
    //outputs
    //output 1: mInitFeedbackData
    int mAdditionOutput;


    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
    }

    public void work()
    {
	mInitFeedbackData = input.popInt();
	mShiftRegisterInput = input.popInt();
	
	 
	output.pushInt(mInitFeedbackData);
	output.pushInt(mShiftRegisterInput);
    }

}//ZeroAdditionFilter  //i.e. the identity filter
    
class ShiftRegisterFilter extends Filter
{
    
    //private register variable:
    int mRegisterNumber;
    int mRegisterContents;
    //inputs:
    int mInitFeedbackData;
    int mDataStream;
    //output:
    int mRegisterOutput;
    int mTerminationCounter;
    
    public ShiftRegisterFilter(int number)
    {
        super(number);
    }
    public void init(int number)
    {
	mRegisterNumber = number;
	mTerminationCounter = 0;
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 2);
	mRegisterContents = 0; //initial value
    }

    public void work()
    {	
	mInitFeedbackData = input.popInt();
	mDataStream = input.popInt();
	mRegisterOutput = mRegisterContents;   //move datastream value into Shift Register
	mRegisterContents = mDataStream;       //move original contents to output
	mTerminationCounter++;
	/*
	if(mTerminationCounter < EncoderInput.mInputLength + 1)
	{
	    System.err.println("Shift Register Contents of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterContents);
	    System.err.println("Shift Register Output of Filter " + mRegisterNumber + " at iteration " + mTerminationCounter + " is " + mRegisterOutput);
	    
	}
	*/
	output.pushInt(mInitFeedbackData);
	output.pushInt(mRegisterOutput);
    }
}//ShiftRegisterFilter

class CrcInputFilter extends Filter
{
    // int[] mdata;
    boolean counter;
    // EncoderInput mencodefile;

    public void init()
    {
	counter = false;
	// mencodefile = new EncoderInput();
	// mdata = new int[mencodefile.mInputLength];
	output = new Channel(Integer.TYPE, 1);
    }

    public void work()
    {
	if(!counter)
	{
	    counter = true;
	    output.pushInt(0);
	}
	else
	{
	    output.pushInt(1);
	}     
    }
	//mdata = mencodefile.readFile();
	   //for (int i = 0; i < mencodefile.mInputLength; i++)
	   //{
	    //System.errprintln("inputting " + mdata[i]);
    //output.pushInt(mdata[i]);
       //}
    

}//CrcInputFilter

class FeedbackEndFilter extends Filter
{
    int mRegisterOutput;
    int mInitFeedbackData;

    public void init()
    {
	input = new Channel(Integer.TYPE, 2);
	output = new Channel(Integer.TYPE, 1);
    }
    
    public void work()
    {
	mInitFeedbackData = input.popInt();
	mRegisterOutput = input.popInt();
	//System.err.println("Shift Register 3 is " + mRegisterOutput);
	output.pushInt(mRegisterOutput);
    }
}//FeedbackEndFilter

class IntPrinter extends Filter {
    public void init()
    {
	input = new Channel(Integer.TYPE, 1);
    }
    
    public void work()
    {
	System.out.println(input.popInt());
    }
}

class CrcFeedbackLoop extends FeedbackLoop
{
    public void init()
    {
	this.setDelay(1);
	this.setJoiner(WEIGHTED_ROUND_ROBIN (1, 1));  //sequence: inputdata, fbdata
	this.setBody(new FeedbackBodyStream());
	this.setSplitter(DUPLICATE ());
	this.setLoop(new Identity(Integer.TYPE));
    }

    public int initPathInt(int index)
    {
	return 0;
    }
}//CrcFeedbackLoop

class FeedbackBodyStream extends Pipeline
{
    public void init()
    {
	this.add(new InitialAdditionFilter());
	this.add(new ShiftRegisterFilter(1));
	this.add(new AdditionFilter());
	this.add(new ShiftRegisterFilter(2));
	this.add(new AdditionFilter()); //x^2
	this.add(new ShiftRegisterFilter(3));
	//x^3
	this.add(new ShiftRegisterFilter(4));
	this.add(new AdditionFilter()); //x^4
	this.add(new ShiftRegisterFilter(5));
	this.add(new AdditionFilter()); //x^5
	this.add(new ShiftRegisterFilter(6));
	//x^6
	this.add(new ShiftRegisterFilter(7));
	this.add(new AdditionFilter()); //x^7
	this.add(new ShiftRegisterFilter(8));
	this.add(new AdditionFilter()); //x^8
	this.add(new ShiftRegisterFilter(9));
	//x^9
	this.add(new ShiftRegisterFilter(10));
	this.add(new AdditionFilter()); //x^10
	this.add(new ShiftRegisterFilter(11));
	this.add(new AdditionFilter()); //x^11
	this.add(new ShiftRegisterFilter(12));
	this.add(new AdditionFilter()); //x^12
	this.add(new ShiftRegisterFilter(13));
	//x^13
	this.add(new ShiftRegisterFilter(14));
	//x^14
	this.add(new ShiftRegisterFilter(15));
	//x^15
	this.add(new ShiftRegisterFilter(16));
	this.add(new AdditionFilter()); //x^16
	this.add(new ShiftRegisterFilter(17));
	//x^17
	this.add(new ShiftRegisterFilter(18));
	//x^18
	this.add(new ShiftRegisterFilter(19));
	//x^19
	this.add(new ShiftRegisterFilter(20));
	//x^20
	this.add(new ShiftRegisterFilter(21));
	//x^21
	this.add(new ShiftRegisterFilter(22));
	this.add(new AdditionFilter()); //x^22
	this.add(new ShiftRegisterFilter(23));
	this.add(new AdditionFilter()); //x^23
	this.add(new ShiftRegisterFilter(24));
	//x^24
	this.add(new ShiftRegisterFilter(25));
	//x^25
	this.add(new ShiftRegisterFilter(26));
	this.add(new AdditionFilter()); //x^26
	this.add(new ShiftRegisterFilter(27));
	//x^27
	this.add(new ShiftRegisterFilter(28));
	//x^28
	this.add(new ShiftRegisterFilter(29));
	//x^29
	this.add(new ShiftRegisterFilter(30));
	//x^30
	this.add(new ShiftRegisterFilter(31));
	//x^31
	this.add(new Identity(Integer.TYPE)); 

	//this.add(new ShiftRegisterFilter(3));
	this.add(new FeedbackEndFilter());
    }
   
}//FeedbackBodyStream

public class CrcEncoder32Test extends StreamIt
{
    public static void main(String args[])
    {
	new CrcEncoder32Test().run(args);
    }

    public void init()
    {
	this.add(new CrcInputFilter());
	this.add(new CrcFeedbackLoop());
	this.add(new IntPrinter());
    }
}//class CrcEncoder32Test





