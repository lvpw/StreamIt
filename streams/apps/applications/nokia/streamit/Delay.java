/**
 * Simple parameterized delay filter.
 **/
import streamit.library.*;

public class Delay extends Pipeline {
    public Delay(int delay) {
	super(delay);
    }
    public void init(int delay) {
	// basically, just add a bunch of unit delay filters
	for (int i=0; i<delay; i++) {
	    this.add(new Delay_one());
	}
    }
}



/** Character Unit delay **/
class Delay_one extends Filter {
    float state;
    public void init() {
	// initial state of delay is 0
	this.state = 0;
	input = new Channel(Float.TYPE,1);
	output = new Channel(Float.TYPE,1);
    }
    public void work() {
	// push out the state and then update it with the input
	// from the channel
	output.pushFloat(this.state);
	this.state = input.popFloat();
    }
}
