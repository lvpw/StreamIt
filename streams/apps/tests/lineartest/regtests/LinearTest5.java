import streamit.library.*;

/**
 * Simple streamit program to test linear filters. This test
 * tests arrays.
 **/
public class LinearTest5 extends StreamIt {
    public static void main(String[] args) {
	StreamIt app = new LinearTest5();
	app.run(args);
    }
    public void init() {
	this.add(new Source());
	this.add(new FloatFilter1());
	this.add(new FloatFilter2());
	this.add(new FloatFilter3());
	this.add(new FloatFilter4());
	this.add(new Sink());
    }
}


/**
 * Single, monotonically increasing source of floating point numbers.
 **/
class Source extends Filter {
    float INCREMENT = (float).01;
    float currentValue = 0;
    public void init() {
	output = new Channel(Float.TYPE, 1);
	this.currentValue = 0;
    }
    public void work() {
	output.pushFloat(this.currentValue);
	this.currentValue += INCREMENT;
    }
}

/**
 * Simple float sink that prints what it gets.
 **/
class Sink extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 1);
    }
    public void work() {
	System.out.println(input.popFloat());
    }
}

/**
 * Just make an array of weights in work function and use them to calcluate the output.
 **/
class FloatFilter1 extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 3);
	output = new Channel(Float.TYPE, 1);
    }
    public void work() {
	float[] weights = new float[4];
	// set up the weights
	weights[0] = (float)1.2;
	weights[1] = (float)4.9;
	weights[2] = (float)3.2;
	weights[3] = (float)4.2;
	    
	float t1 = input.popFloat();
	float t2 = input.popFloat();
	float t3 = input.popFloat();

	// now, push out weighted combos of t1,t2,t3 based
	// on the array weights.

	// col3 = [3.2 4.9 1.2]+[4.2]
	output.pushFloat(t1*weights[0] +
			 weights[1]*t2 +
			 t3*weights[2] +
			 weights[3]);
    }
}


/**
 * Store intermediate values in arrays this time, along with using the weights array
 **/
class FloatFilter2 extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 3);
	output = new Channel(Float.TYPE, 3);
    }
    public void work() {
	float[] weights = new float[4];
	// set up the weights
	weights[0] = 1;
	weights[1] = 2;
	weights[2] = 3;
	weights[3] = 4;
	    
	float t1 = input.popFloat();
	float t2 = input.popFloat();
	float t3 = input.popFloat();
	
	// calculate the output into the intermediate array and then
	// push its values
	float[] intermediate = new float[3];
	intermediate[0] = (t1 +
			   weights[0]*weights[1]*t2 +
			   weights[2]*t3*weights[3]);

	// intermediate[0] -> [12 2 1]+[0]

	intermediate[1] = intermediate[0];
	// intermediate[1] -> [12 2 1]+[0]
	intermediate[1] += intermediate[1];
	// intermediate[1] -> [24 4 2]+[0]
	intermediate[1] = 3 + intermediate[1];
	// intermediate[1] -> [24 4 2]+[3]

	intermediate[2] = t1 + t2 + 5*t3 + 7;
	// intermediate[2] -> [5 1 1 ]+[7]	
	intermediate[2] += intermediate[1] + intermediate[0] + t3;
	// intermediate[2] -> [42 7 4]+[10]

	// push out the values we just calculated
	// col3 = [12 2 1]+[0] 
	output.pushFloat(intermediate[0]);
	// col2 = [24 4 2]+[3]
	output.pushFloat(intermediate[1]);
	// col1 = [42 7 4]+[10]
	output.pushFloat(intermediate[2]);
    }
}


/** test out arrays with initializers **/
class FloatFilter3 extends Filter {
    public void init() {
	input = new Channel(Float.TYPE, 3);
	output = new Channel(Float.TYPE, 3);
    }
    public void work() {
	float[] weights = new float[4];
	for (int i=0; i<4; i++) {
	    weights[i] = i+1;
	}
	float t1 = input.popFloat();
	float t2 = input.popFloat();
	float t3 = input.popFloat();

	// col3 = [4 3 2]+[1]
	output.pushFloat(t3*weights[3] +
			 t2*weights[2] +
			 t1*weights[1] +
			 weights[0]);

	// make sure that the semantics of arrays starting with all zeros
	// in maintained.
	float[] zeroarray = new float[2];
	zeroarray[0] = 0;
	t3 = zeroarray[0];
	// col2 = [0 1 1]+[0]
	output.pushFloat(t1+t2+t3);

	// make sure, for that matter, that normal variables are initialized to zero
	//float newvar; // you can't acutally do this because the JLS says that you have to
	// initialize all variables before use.
	float newvar=0;
	t2 = newvar;
	// col1 = [0 0 1]+[0]
	output.pushFloat(t1+t2+t3);
    }
}



/**
 * Test a simple FIR type filter with weights that are
 * calculated both in init and weights that are calculated
 * in the actual work function;
 **/
class FloatFilter4 extends Filter {
    final int SIZE = 5;
    float[] weightField = new float[SIZE];
    
    public void init() {
	input = new Channel(Float.TYPE, SIZE);
	output = new Channel(Float.TYPE, 2);
	for (int i=0; i<SIZE; i++) {
	    weightField[i] = i*2;
	}
    }
    public void work() {
	float[] weights;

	// first, use the weights calculated in the init function
	float sum=0;
	for (int i=0; i<SIZE; i++) {
	    sum += input.peekFloat(i)*weightField[i];
	}

	// col2=[8 6 4 2 0]+0;
	output.pushFloat(sum);

	// now calculate weights in the work function
	weights = new float[SIZE];
	for (int i=0; i<SIZE; i++) {
	    weights[i] = i+3;
	}

	// calculate the sum
	sum = 0;
	for (int i=0; i<SIZE; i++) {
	    sum += weights[i]*input.peekFloat(i);
	}
	
	// col1 = [7 6 5 4 3]+0
	output.pushFloat(sum);

	
	// finally, pop the appropriate number from the buffer
	for (int i=0; i<SIZE; i++) {
	    input.popFloat();
	}

	
    }
}
