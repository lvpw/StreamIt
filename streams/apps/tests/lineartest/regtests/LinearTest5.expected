FloatFilter1_1.matrix
[[3.200000047683716+0.0i]
 [4.900000095367432+0.0i]
 [1.2000000476837158+0.0i]]
FloatFilter1_1.vector
[[4.199999809265137+0.0i]]
FloatFilter2_1.matrix
[[42.0+0.0i 24.0+0.0i 12.0+0.0i]
 [7.0+0.0i 4.0+0.0i 2.0+0.0i]
 [4.0+0.0i 2.0+0.0i 1.0+0.0i]]
FloatFilter2_1.vector
[[10.0+0.0i 3.0+0.0i 0.0+0.0i]]
FloatFilter3_1.matrix
[[0.0+0.0i 0.0+0.0i 4.0+0.0i]
 [0.0+0.0i 1.0+0.0i 3.0+0.0i]
 [1.0+0.0i 1.0+0.0i 2.0+0.0i]]
FloatFilter3_1.vector
[[0.0+0.0i 0.0+0.0i 1.0+0.0i]]
FloatFilter4_1.matrix
[[7.0+0.0i 8.0+0.0i]
 [6.0+0.0i 6.0+0.0i]
 [5.0+0.0i 4.0+0.0i]
 [4.0+0.0i 2.0+0.0i]
 [3.0+0.0i 0.0+0.0i]]
FloatFilter4_1.vector
[[0.0+0.0i 0.0+0.0i]]
LinearWrapper_1.matrix
[[5184.00007724762+0.0i 4377.600065231323+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [7938.000154495239+0.0i 6703.2001304626465+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [1944.0000772476196+0.0i 1641.6000652313232+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [864.0000128746033+0.0i 729.6000108718872+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [1323.0000257492065+0.0i 1117.2000217437744+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [324.00001287460327+0.0i 273.6000108718872+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [464.0000069141388+0.0i 390.40000581741333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [710.5000138282776+0.0i 597.8000116348267+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [174.0000069141388+0.0i 146.40000581741333+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i]
 [499.20000743865967+0.0i 76.80000114440918+0.0i 5913.600088119507+0.0i 6758.400100708008+0.0i 0.0+0.0i 0.0+0.0i]
 [764.4000148773193+0.0i 117.60000228881836+0.0i 9055.200176239014+0.0i 10348.800201416016+0.0i 0.0+0.0i 0.0+0.0i]
 [187.20000743865967+0.0i 28.80000114440918+0.0i 2217.600088119507+0.0i 2534.400100708008+0.0i 0.0+0.0i 0.0+0.0i]
 [83.20000123977661+0.0i 12.800000190734863+0.0i 985.6000146865845+0.0i 1126.400016784668+0.0i 0.0+0.0i 0.0+0.0i]
 [127.40000247955322+0.0i 19.600000381469727+0.0i 1509.200029373169+0.0i 1724.800033569336+0.0i 0.0+0.0i 0.0+0.0i]
 [31.20000123977661+0.0i 4.800000190734863+0.0i 369.6000146865845+0.0i 422.40001678466797+0.0i 0.0+0.0i 0.0+0.0i]
 [41.600000619888306+0.0i 6.400000095367432+0.0i 537.6000080108643+0.0i 614.4000091552734+0.0i 0.0+0.0i 0.0+0.0i]
 [63.70000123977661+0.0i 9.800000190734863+0.0i 823.2000160217285+0.0i 940.8000183105469+0.0i 0.0+0.0i 0.0+0.0i]
 [15.600000619888306+0.0i 2.4000000953674316+0.0i 201.60000801086426+0.0i 230.40000915527344+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 4185.6000623703+0.0i 2380.8000354766846+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 6409.200124740601+0.0i 3645.600070953369+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 1569.6000623703003+0.0i 892.8000354766846+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 697.60001039505+0.0i 396.80000591278076+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 1068.2000207901+0.0i 607.6000118255615+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 261.60001039505005+0.0i 148.80000591278076+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 374.40000557899475+0.0i 211.20000314712524+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 573.3000111579895+0.0i 323.4000062942505+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 140.40000557899475+0.0i 79.20000314712524+0.0i 0.0+0.0i 0.0+0.0i]
 [0.0+0.0i 0.0+0.0i 115.20000171661377+0.0i 0.0+0.0i 5875.200087547302+0.0i 5990.400089263916+0.0i]
 [0.0+0.0i 0.0+0.0i 176.40000343322754+0.0i 0.0+0.0i 8996.400175094604+0.0i 9172.800178527832+0.0i]
 [0.0+0.0i 0.0+0.0i 43.20000171661377+0.0i 0.0+0.0i 2203.2000875473022+0.0i 2246.400089263916+0.0i]
 [0.0+0.0i 0.0+0.0i 19.200000286102295+0.0i 0.0+0.0i 979.200014591217+0.0i 998.4000148773193+0.0i]
 [0.0+0.0i 0.0+0.0i 29.40000057220459+0.0i 0.0+0.0i 1499.400029182434+0.0i 1528.8000297546387+0.0i]
 [0.0+0.0i 0.0+0.0i 7.200000286102295+0.0i 0.0+0.0i 367.20001459121704+0.0i 374.40001487731934+0.0i]
 [0.0+0.0i 0.0+0.0i 9.600000143051147+0.0i 0.0+0.0i 528.0000078678131+0.0i 537.6000080108643+0.0i]
 [0.0+0.0i 0.0+0.0i 14.700000286102295+0.0i 0.0+0.0i 808.5000157356262+0.0i 823.2000160217285+0.0i]
 [0.0+0.0i 0.0+0.0i 3.6000001430511475+0.0i 0.0+0.0i 198.0000078678131+0.0i 201.60000801086426+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 3187.200047492981+0.0i 384.0000057220459+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 4880.400094985962+0.0i 588.0000114440918+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 1195.200047492981+0.0i 144.0000057220459+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 531.2000079154968+0.0i 64.00000095367432+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 813.4000158309937+0.0i 98.00000190734863+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 199.20000791549683+0.0i 24.000000953674316+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 284.8000042438507+0.0i 32.00000047683716+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 436.1000084877014+0.0i 49.000000953674316+0.0i]
 [0.0+0.0i 0.0+0.0i 0.0+0.0i 0.0+0.0i 106.80000424385071+0.0i 12.000000476837158+0.0i]]
LinearWrapper_1.vector
[[9642.999574661255+0.0i 7559.599666595459+0.0i 17415.39923477173+0.0i 15589.99931526184+0.0i 15426.599321365356+0.0i 10838.399522781372+0.0i]]
